generate.go

```diff
@@ -474,19 +529,42 @@ type Record struct {
 	Container *metadata.Container
 }
 
-func (mf rMetaFetcher) GetService(svcName string, stackName string) (*metadata.Service, error) {
-	svcs, err := mf.metadataClient.GetServices()
+func (mf rMetaFetcher) GetService(link string) (metadata.Service, error) {
+	splitSvcName := strings.Split(link, "/")
+	var linkedService metadata.Service
+	var err error
+
+	regionName, err := mf.GetRegionName()
 	if err != nil {
-		return nil, err
+		return linkedService, err
 	}
-	var service metadata.Service
-	for _, svc := range svcs {
-		if strings.EqualFold(svc.Name, svcName) && strings.EqualFold(svc.StackName, stackName) {
-			service = svc
-			break
+	regionName = strings.TrimSuffix(regionName, "\"")
+	regionName = strings.TrimPrefix(regionName, "\"")
```

Comment:
is line 542 a duplicate of 541?

Reply:
Doing suffix, prefix separately. Will do it in one line (also, shifting this to metadata function)